src/io/bnx.rs

```diff
@@ -0,0 +1,1486 @@
+                Record::read_label_for_channel(label_lines[0], label_lines[1], label_lines[2])?;
+            Ok((channel_1, Vec::new()))
+        } else if num_lines == 6 {
+            let channel_1 =
+                Record::read_label_for_channel(label_lines[0], label_lines[2], label_lines[4])?;
+            let channel_2 =
+                Record::read_label_for_channel(label_lines[1], label_lines[3], label_lines[5])?;
+            Ok((channel_1, channel_2))
+        } else {
+            bail!(Error::InvalidBnxRecord(num_lines));
+        }
+    }
+
+    /// Reads label information for one channel.
+    fn read_label_for_channel(
+        pos_line: &str,
+        snr_line: &str,
+        intensity_line: &str,
+    ) -> Result<Vec<Label>> {
+        let mut labels: Vec<Label> = Vec::new();
+        let pos_line = match RE_POS_LINE.is_match(pos_line) {
+            true => &pos_line[2..],
+            false => {
+                bail!(Error::InvalidLabelLine("LabelChannel"))
+            }
+        };
+        let snr_line = match RE_SNR_LINE.is_match(snr_line) {
+            true => &snr_line[5..],
+            false => {
+                bail!(Error::InvalidLabelLine("Label SNR"))
+            }
+        };
+        let intensity_line = match RE_INTENSITY_LINE.is_match(intensity_line) {
+            true => &intensity_line[5..],
+            false => bail!(Error::InvalidLabelLine("Label Intensity")),
+        };
+        let pos_count = pos_line.matches('\t').count();
+        let snr_count = snr_line.matches('\t').count();
+        let intensity_count = intensity_line.matches('\t').count();
+        if pos_count - 1 != snr_count || snr_count != intensity_count {
+            bail!(Error::InvalidBnxLabel)
+        }
+        let snr_line = &(snr_line.to_string() + "\t0.0");
+        let intensity_line = &(intensity_line.to_string() + "\t0.0");
+        for entry in izip!(
+            pos_line.split('\t'),
+            snr_line.split('\t'),
+            intensity_line.split('\t')
+        ) {
+            labels.push(Label {
+                pos: entry
+                    .0
+                    .parse()
+                    .context(Error::InvalidType("LabelPosition", "f64 float"))?,
+                snr: entry
+                    .1
+                    .parse()
+                    .context(Error::InvalidType("QualityScore (SNR)", "f32 float"))?,
+                intensity: entry
+                    .2
+                    .parse()
+                    .context(Error::InvalidType("QualityScore (Intensity)", "f32 float"))?,
+            });
+        }
+        Ok(labels)
+    }
+
+    /// Returns molecule ID (BNX ID).
+    pub fn id(&mut self) -> u32 {
+        self.id
+    }
+
+    /// Returns labels from channel 1.
+    pub fn channel_1(&mut self) -> Vec<Label> {
+        self.channel_1.clone()
+    }
+
+    /// Return labels from channel 2.
+    pub fn channel_2(&mut self) -> Vec<Label> {
+        self.channel_2.clone()
+    }
```

Comment:
Can be deleted and handled by the Getters/CopyGetters derive approach (see above).